modules/mod_ginger_edit/controller/controller_ginger_edit.erl

```diff
@@ -0,0 +1,86 @@
+-module(controller_ginger_edit).
+-author("Loetie Kruger <loetie@driebit.nl>").
+
+-export([
+    resource_exists/2,
+    previously_existed/2,
+    moved_temporarily/2,
+    is_authorized/2,
+    html/1
+]).
+
+-include_lib("controller_html_helper.hrl").
+
+%% @doc Check if the id in the request (or dispatch conf) exists.
+resource_exists(ReqData, Context) ->
+    Context1  = ?WM_REQ(ReqData, Context),
+    ContextQs = z_context:ensure_qs(Context1),
+    try
+        Id = z_controller_helper:get_id(ContextQs),
+        maybe_exists(Id, ContextQs)
+    catch
+        _:_ -> ?WM_REPLY(false, ContextQs)
+    end.
```

Comment:
That is indeed better. Did this now for all functions that were possible. 
Did not do for:
- resource_exist: this is not possible since this function includes the redirect logic which we dont want to use. 
- maybe_exists function is not exported in controller_page.